plugins/module_utils/nae.py

```diff
@@ -1020,15 +1020,40 @@ def send_pre_change_payload(self):
 
         self.result['Result'] = "Pre-change analysis %(name)s successfully created." % self.params
 
+    def check_existing(self):
+        try:
+            form = json.loads(self.params.get('form'))
+        except Exception:
+            self.module.fail_json(msg='The form can not be loaded properly')
+        if form is None:
+            self.module.fail_json(msg='The form is empty')
+        elif form.get('name') is None:
+            self.module.fail_json(msg='The name should not be empty')
+        obj, type_map = self.query_compliance_object(form.get('name'))
+        return obj, form
+
     def new_object_selector(self):
         self.params['fabric_uuid'] = self.getFirstAG().get("uuid")
-        url = 'https://%(host)s:%(port)s/nae/api/v1/event-services/'\
-              'assured-networks/%(fabric_uuid)s/model/aci-policy/'\
-              'compliance-requirement/object-selectors' % self.params
+        obj, form = self.check_existing()
+        if obj != []:
+            url = 'https://{0}:{1}/nae/api/v1/event-services/'\
+                  'assured-networks/{2}/model/aci-policy/'\
+                  'compliance-requirement/object-selectors/{3}'.format(self.params.get('host'),
+                                                                       self.params.get('port'),
+                                                                       self.params.get('fabric_uuid'),
+                                                                       str(obj[0].get('uuid')))
+            method = 'PUT'
+            form['uuid'] = obj[0].get('uuid')
+            self.params['form'] = json.dumps(form)
```

Comment:
should we check that UUID is not None?

Reply:
I suppose that UUID will always be included in obj which we query from the nae

Reply:
Ok